### PR TITLE
Feature/pupils not taking check refactoring

### PR DESCRIPTION
**Pull request by pris54**
Refactoring and unit tests related to feature 14131 "Pupils not taking the check"

### CODE REVIEW

**Comment by GuyHarwood** [comment]
@pris54 instead of just adding 'WIP' to your commits, can you write a brief description of the changes you made.  It's really helpful when in situations where we might need to comb back through commits, as it tells a meaningful story.  A wise man once said...

> "imagine that you are handing over your repo to a psychopath"

**Comment by pris54** [comment]
@GuyHarwood HAHAHA, OK, I will, I'm lazy sometimes.

File: admin/spec/service/pupils-not-taking-check.service.spec.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      service = setupService(pupilsWithReasonsMock)
+    })
+
+    it('should return a list of pupils with reasons (happy path)', async (done) => {
+      try {
+        const pupilsWithReasons = await service.fetchPupilsWithReasons(9991001)
+        expect(pupilsWithReasons).toEqual(pupilsWithReasonsMock)
+        done()
+      } catch (error) {
+        done('Error found while testing fetchPupilsWithReasons')
```
Comment (jon-shipley) [inline admin/spec/service/pupils-not-taking-check.service.spec.js]: This style no longer works with jasmine.  Instead you have to do:
`expect(true).toBe(false)`
if you want to guard against it throwing.

File: admin/services/pupils-not-taking-check.service.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+/** @namespace */
+
+const pupilsNotTakingCheckService = {
+
+  /**
+   * Fetch pupils with reasons to not take the check.
+   * @param schoolId
+   * @returns {Promise.<*>}
+   */
+  fetchPupilsWithReasons: async (schoolId) => {
```
Comment (jon-shipley) [inline admin/services/pupils-not-taking-check.service.js]: This should really be in the data service layer

File: admin/services/pupils-not-taking-check.service.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    .sort('lastName')
+    if (pupilsWithReasons.length > 0) {
+      return pupilsWithReasons
+    }
+  },
+  /**
+   * Sort columns by last name asc/desc.
+   * @param sortField
+   * @param sortDirection
+   * @returns {Promise.<{htmlSortDirection: Array, arrowSortDirection: Array}>}
```
Comment (jon-shipley) [inline admin/services/pupils-not-taking-check.service.js]: I don't see a Promise being returned

Comment (pris54) [reply admin/services/pupils-not-taking-check.service.js]: Indeed, that's wrong. Fixed in next commit.

File: admin/spec/service/pupils-not-taking-check.service.spec.js (hunk 1 above)
Comment (pris54) [reply admin/spec/service/pupils-not-taking-check.service.spec.js]: Fixed in next commit.

**Pull request merged**